distribution/std-lib/Base/src/Time/Time_Of_Day.enso

```diff
@@ -0,0 +1,119 @@
+from Base import all
+import Base.Time
+import Base.Time.Date
+import Base.Time.Duration
+import Base.Time.Zone
+
+polyglot java import java.time.Instant
+polyglot java import java.time.LocalTime
+polyglot java import java.time.format.DateTimeFormatter
+polyglot java import org.enso.base.Time_Utils
+
+type Time_Of_Day
+
+    ## `Time_Of_Day` is a date-time object that represents a time, often viewed
+       as hour-minute-second. Time is represented to nanosecond precision. For
+       example, the value "13:45.30.123456789" can be stored in a `Time_Of_Day`.
+    type Time_Of_Day internal_local_time
+
+    ## Get the hour of day field.
+    get_hour : Integer
+    get_hour = this . internal_local_time . getHour []
+
+    ## Get the minute of hour field.
+    get_minute : Integer
+    get_minute = this . internal_local_time . getMinute []
+
+    ## Get the second of minute field.
+    get_second : Integer
+    get_second = this . internal_local_time . getSecond []
+
+    ## Get the nanosecond of second field.
+    get_nano : Integer
+    get_nano = this . internal_local_time . getNano []
+
+    ## Get the nanosecond of second field.
+    get_nanosecond : Integer
+    get_nanosecond = this.get_nano
+
+    ## Extracts the time as the number of seconds, from 0 to 24 * 60 * 60 - 1.
+    to_seconds : Integer
```

Comment:
This is a great use of `to`, so I decided to explain my rationale. `to` should be used for conversion. All of the `to` methods I commented in `Time` are logically getter-like, not converter-like. I.e. a `Time` _has_ a `Date`, but this here is clearly a conversion.